runtime/caml/fail.h

```diff
@@ -139,6 +139,18 @@ Caml_inline value caml_get_value_or_raise (struct caml_result_private result)
     return result.data;
 }
 
+#ifdef CAML_INTERNALS
+// internals only, provided for backward-compatibility
+Caml_inline value caml_result_get_encoded_exception(
+  struct caml_result_private result)
```

Comment:
Is this function supposed to be part of the puvlic API of `caml_result`?

Reply:
No, it is CAML_INTERNALS only. It is only useful in transition scenarios, but I suspect that we may want to use it in more parts of the runtime occasionally (which is why factored it out of `caml_process_pending_actions_exn`).